warehouse/oidc/forms/activestate.py

```diff
@@ -0,0 +1,212 @@
+            )
+            raise wtforms.validators.ValidationError(
+                _("Unexpected error from ActiveState. Try again")
+            )
+    except (requests.Timeout, requests.ConnectionError):
+        sentry_sdk.capture_message("Connection error from ActiveState API")
+        raise wtforms.validators.ValidationError(
+            _("Unexpected error from ActiveState. Try again")
+        )
+    # Graphql reports it's errors within the body of the 200 response
+    try:
+        response_json = response.json()
+        errors = response_json.get("errors")
+        if errors:
+            sentry_sdk.capture_message(
+                f"Unexpected error from ActiveState API: {errors}"
+            )
+            raise wtforms.validators.ValidationError(
+                _("Unexpected error from ActiveState. Try again")
+            )
+
+        return response_handler(response_json)
+    except requests.exceptions.JSONDecodeError:
+        sentry_sdk.capture_message(
+            f"Unexpected error from ActiveState API: {response.content!r}"
+        )
+        raise wtforms.validators.ValidationError(
+            _("Unexpected error from ActiveState. Try again")
+        )
+
+
+class ActiveStatePublisherBase(forms.Form):
+    __params__ = ["organization", "project", "actor"]
+
+    organization = wtforms.StringField(
+        validators=[
+            wtforms.validators.InputRequired(
+                message=_("Specify ActiveState organization name"),
+            ),
+            wtforms.validators.Regexp(
+                _VALID_ORG_URL_NAME_AND_ACTOR_NAME,
+                message=_("Invalid ActiveState organization name"),
+            ),
+            _no_double_dashes,
+            _no_leading_or_trailing_dashes,
+        ]
+    )
+
+    project = wtforms.StringField(
+        validators=[
+            wtforms.validators.InputRequired(
+                message=_("Specify ActiveState project name")
+            ),
+            wtforms.validators.Regexp(
+                _VALID_PROJECT_NAME,
+                message=_("Invalid ActiveState project name"),
+            ),
+            wtforms.validators.Regexp(
+                _VALID_PROJECT_NAME,
+                message=_("Invalid ActiveState project name"),
+            ),
+            _no_double_dashes,
+            _no_leading_or_trailing_dashes,
+        ]
+    )
+
+    actor = wtforms.StringField(
+        validators=[
+            wtforms.validators.InputRequired(
+                message=("Specify the ActiveState actor username")
+            ),
+            wtforms.validators.Regexp(
+                _VALID_ORG_URL_NAME_AND_ACTOR_NAME,
+                message=("Invalid ActiveState username"),
+            ),
+            _no_double_dashes,
+            _no_leading_or_trailing_dashes,
+        ]
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
I think this can be removed outright, since we're not specializing the ctor in any way:

```suggestion
```